worker/uploader.go

```diff
@@ -134,16 +138,49 @@ outer:
 				err: err,
 			}:
 			}
+		}
+	}
+}
 
-			// track the error, ignore gracefully closed streams and canceled overdrives
-			canceledOverdrive := req.done() && req.overdrive && err != nil
-			if !canceledOverdrive && !isClosedStream(err) {
-				u.trackSectorUpload(err, elapsed)
-			} else {
-				u.logger.Debugw("not tracking sector upload metric", zap.Error(err))
-			}
+func handleSectorUpload(uploadErr error, uploadDuration, totalDuration time.Duration, overdrive bool, logger *zap.SugaredLogger) (success bool, failure bool, uploadEstimateMS float64, uploadSpeedBytesPerMS float64) {
```

Comment:
`success` and `failure` seem to always have opposite values as you'd expect except for when the upload "doesn't count". Maybe a `penalize` bool would be better?

Reply:
`success` and `failure` are only used to adjust `consecutiveFailures`, adding a `penalize` bool doesn't really help I think because there's 3 actions we can trigger: reset consecutive failures, increment or do nothing at all, so I need the two bools :/